api/lib/opentelemetry/instrumentation/registry.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    # The instrumentation Registry contains information about instrumentation
+    # adapters available and facilitates discovery, installation and
+    # configuration. This functionality is primarily useful for SDK
+    # implementors.
+    class Registry
+      def initialize
+        @lock = Mutex.new
+        @adapters = []
+      end
+
+      # @api private
+      def register(adapter)
+        @lock.synchronize do
+          @adapters << adapter
+        end
+      end
+
+      # Lookup an adapter definition by name. Returns nil if +adapter_name+
+      # is not found.
+      #
+      # @param [String] adapter_name A stringified class name for an adapter
+      # @return [Adapter]
+      def lookup(adapter_name)
+        @lock.synchronize do
+          find_adapter(adapter_name)
+        end
+      end
+
+      # Install the specified adapters with optionally specified configuration.
+      #
+      # @param [Array<String>] adapter_names An array of adapter names to
+      #   install
+      # @param [optional Hash<String, Hash>] adapter_config_map A map of
+      #   adapter_name to config. This argument is optional and config can be
+      #   passed for as many or as few adapters as desired.
+      def install(adapter_names, adapter_config_map = {})
+        @lock.synchronize do
+          adapter_names.each do |adapter_name|
```

Comment:
You could wrap `adapter_names` in `Array()` to allow this method to accept a single name or an array.

```ruby
def install(adapter_names, adapter_config_map = {})
  @lock.synchronize do
    Array(adapter_names).each do |adapter_name|

  # ...
end
```

Reply:
Similar to my comment above. This is an API method that will be used by the SDK and the SDK will always pass in an array. It's a simple change to make if you think there's a use for it though.

Reply:
@mwear Gotcha. I'm good with this as is - especially given that it's internal. Thank you.